src/fleet/azext_fleet/tests/latest/test_validators.py

```diff
@@ -129,5 +139,75 @@ def test_empty_identity_id(self):
 
         self.assertIsNone(validators.validate_assign_identity(namespace))
 
+class TestValidateUpgradeType(unittest.TestCase):
+    def test_valid_upgrade_type(self):
+        valid_upgrade_type = "NodeImageOnly"
+        namespace = UpgradeTypeNamespace(valid_upgrade_type)
+
+        self.assertIsNone(validators.validate_upgrade_type(namespace))
+
+    def test_invalid_upgrade_type(self):
+        invalid_upgrade_type = "invalid_upgrade_type"
+        namespace = UpgradeTypeNamespace(invalid_upgrade_type)
+        err = ("--upgrade-type must be set to 'Full' or 'NodeImageOnly'")
+
+        with self.assertRaises(CLIError) as cm:
+            validators.validate_upgrade_type(namespace)
+        self.assertEqual(str(cm.exception), err)
+
+    def test_empty_upgrade_type(self):
+        empty_upgrade_type = ""
+        namespace = UpgradeTypeNamespace(empty_upgrade_type)
+        err = ("--upgrade-type must be set to 'Full' or 'NodeImageOnly'")
+
+        with self.assertRaises(CLIError) as cm:
+            validators.validate_upgrade_type(namespace)
+        self.assertEqual(str(cm.exception), err) 
+
+    def test_none_upgrade_type(self):
+        none_upgrade_type = None
+        namespace = UpgradeTypeNamespace(none_upgrade_type)
+        err = ("--upgrade-type must be set to 'Full' or 'NodeImageOnly'")
+
+        with self.assertRaises(CLIError) as cm:
+            validators.validate_upgrade_type(namespace)
+        self.assertEqual(str(cm.exception), err)
+
+class TestValidateNodeImageSelection(unittest.TestCase):
+    def test_valid_node_image_selection(self):
+        valid_node_image_selection = "Latest"
+        namespace = NodeImageSelectionNamespace(valid_node_image_selection)
+
+        self.assertIsNone(validators.validate_node_image_selection(namespace))
+
+    def test_invalid_node_image_selection(self):
+        invalid_node_image_selection = "invalid_node_image_selection"
+        namespace = NodeImageSelectionNamespace(invalid_node_image_selection)
+        err = ("--node-image-selection must be set to 'Latest' or 'Consistent'")
```

Comment:
presumably these tests are currently failing and will pass once the error string is updated?

Reply:
The validator has been removed as there is an `arg_type=get_enum_type function now used on the parameters.
```
arg_type=get_enum_type(['Latest', 'Consistent'])
```